include/seqan3/io/sequence_file/format_sam.hpp

```diff
@@ -0,0 +1,346 @@
+                                      })
+                                    | view::char_to<value_type_t<seq_type>>,      // convert to actual target alphabet
+                           ranges::back_insert_iterator{sequence});
+
+              sequence_size_after = ranges::size(sequence);
+            }
+          else
+          {
+            auto it = ranges::begin(seq_view);
+            auto it_end = ranges::end(seq_view);
+            while (it != it_end)
+            {
+                ++it;
+                ++sequence_size_after;
+            }
+          }
+          detail::consume(stream_view | ranges::view::take_while(is_tab));
+
+          /* Qualities */
+          if constexpr (seq_qual_combined)
+          {
+              // seq_qual field implies that they are the same variable
+              assert(std::addressof(sequence) == std::addressof(qualities));
+              ranges::copy(stream_view | view::char_to<typename value_type_t<qual_type>::quality_alphabet_type>,
+                           ranges::begin(qualities) + sequence_size_before);
+          }
+          else if constexpr (!detail::decays_to_ignore_v<qual_type>)
+          {
+            ranges::copy(stream_view | view::take_until_or_throw(is_cntrl) //TODO: What if we have tags?
+                        | view::char_to<value_type_t<qual_type>>,
+                        ranges::back_insert_iterator{qualities});
+          }
+          else
+          {
+              detail::consume(stream_view | ranges::view::take_while(!is_cntrl));
+          }
+
+
+          // make sure "buffer at end" implies "stream at end"
+          if ((std::istreambuf_iterator<char>{stream} == std::istreambuf_iterator<char>{}) &&
+              (!stream.eof()))
+          {
+              stream.get(); // triggers error in stream and sets eof
+          }
+
+          if ((std::istreambuf_iterator<char>{stream} == std::istreambuf_iterator<char>{}) &&
+              (!stream.eof()))
+          {
+              stream.get(); // triggers error in stream and sets eof
+          }
+
+          // Jump over the new line character in the end
+          detail::consume(stream_view | ranges::view::take_while(is_cntrl));
+        }
+        else{
+             detail::consume(stream_view | view::take_line_or_throw);
+        }
+
+
+
+    }
+
+    //!\copydoc sequence_file_output_format_concept::write
+    template <typename stream_type,     // constraints checked by file
+              typename seq_type,        // other constraints checked inside function
+              typename id_type,
+              typename qual_type>
+    void write(stream_type                     & stream,
+               sequence_file_output_options const & options,
+               seq_type                       && sequence,
+               id_type                        && id,
+               qual_type                      && qualities)
+    {
+        ranges::ostreambuf_iterator stream_it{stream};
+
```

Comment:
Note that you use a lot of functionality for the types that might not compile if you e.g. give an integer to `seq_type`. Therefore the types should be restricted with concepts. With @h-2 I discussed that this should be done as static asserts to give the user a helpful error message.
```
static_assert(std::ForwardRange<seq_type> && alphabet_concept<value_type_t<seq_type>>, 
              "The sequence must model std::ForwardRange and its value type must model seqan3::alphabet_concept. ")